app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlKeyAutogeneratorObserver.php

```diff
@@ -32,5 +32,6 @@ public function execute(\Magento\Framework\Event\Observer $observer)
         /** @var Product $product */
         $product = $observer->getEvent()->getProduct();
         $product->setUrlKey($this->productUrlPathGenerator->getUrlKey($product));
+        $product->setUrlPath($this->productUrlPathGenerator->getUrlPath($product));
```

Comment:
The issue mentioned using `[product-suffix-from-setting]` to build the path.  Should this not be `getUrlPathWithSuffix`?